src/audio/miniaudio/ma_context.cpp

```diff
@@ -0,0 +1,43 @@
+#include "ma_context.hpp"
+
+namespace AbaddonClient::Audio::Miniaudio {
+
+MaContext::MaContext(ContextPtr &&context) noexcept :
+    m_context(std::move(context)) {}
+
+std::optional<MaContext> MaContext::Create(ma_context_config &&config, ConstSlice<ma_backend> backends) noexcept {
+    ContextPtr context = ContextPtr(new ma_context, &ma_context_uninit);
+
+    const auto result = ma_context_init(backends.data(), backends.size(), &config, context.get());
+    if (result != MA_SUCCESS) {
+        spdlog::get("audio")->error("Failed to create context: {}", ma_result_description(result));
+        return std::nullopt;
+    }
+
+    return std::make_optional<MaContext>(std::move(context));
```

Comment:
You want it to be like this?
```cpp
return MaContext(std::move(context));
```

`std::make_optional` constructs the value in-place so that saves a move

Reply:
you mean as a result of it being returned? im fairly certain this would behave exactly the same:
```cpp
return std::move(context);
```

Reply:
Looks like it does, will go with that